doc/book/in-depth-guide/models-and-servicemanager.md

```diff
@@ -523,7 +522,7 @@ return [
 ];
 ```
 
-This aliases `PostRepositoryInterface` to our `PostRepository` implementation,
+This aliases `PostRepositoryInterface` to our `PostRepository` implementation [[Could you explain why this is done? Aliasing usually means using the alias in place of the aliased. How do you use the interface in place of the class?]],
```

Comment:
In zend-servicemanager, when you request a service by an alias, you then get the target of that alias. This is true in PHP as well; if you look at [class_alias](http://php.net/class_alias), when you request a class by `$alias`, you'll get an instance of the `$original` class (unlike zend-servicemanager, however, `$instance instanceof $alias` will _also_ evaluate to true!). In the case of zend-servicemanager, when you request by the alias name, you'll get the service it maps to.

Think of it more like a symlink. When you call `ln -s <target> <link>`, if you access `<link>` later, you'll actually be accessing `<target>`.

We often alias an interface to an implementation service, as that allows the user to indicate they want _an implementation of the interface_, but do not care _which implementation_.

Reply:
Thanks for explaining this, it makes more sense to me now. I will incorporate an answer. Obviously feel free to edit as you see fit.